modules/substrate/src/lib.rs

```diff
@@ -96,15 +105,18 @@ decl_storage! {
 				.initial_header
 				.clone()
 				.expect("An initial header is needed");
+			let initial_hash = initial_header.hash();
 
-			<BestHeader<T>>::put(initial_header.hash());
-			<BestFinalized<T>>::put(initial_header.hash());
+			<BestHeight<T>>::put(initial_header.number());
+			<BestHeaders<T>>::put(vec![initial_hash]);
+			<BestFinalized<T>>::put(initial_hash);
 			<ImportedHeaders<T>>::insert(
-				initial_header.hash(),
+				initial_hash,
 				ImportedHeader {
 					header: initial_header,
 					requires_justification: false,
 					is_finalized: true,
+					signal_hash: None,
```

Comment:
Are you sure it should be `None`. I think it should be `Some(initial_hash)` if `config.first_scheduled_change.is_some()`

Reply:
Also probably worth to add something like `assert!(first_scheduled_change.height > initial_header.number)`

Reply:
You're right, good catch